packages/app-desktop/services/spellChecker/SpellCheckerServiceDriverNative.ts

```diff
@@ -17,43 +16,25 @@ export default class SpellCheckerServiceDriverNative extends SpellCheckerService
 		return this.session().availableSpellCheckerLanguages;
 	}
 
-	// Language can be set to '' to disable spell-checking
-	public setLanguage(v: string) {
+	// Language can be set to [] to disable spell-checking
+	public setLanguage(v: string[]) {
 		// If we pass an empty array, it disables spell checking
 		// https://github.com/electron/electron/issues/25228
-		if (!v) {
+		if (v.length == 0) {
 			this.session().setSpellCheckerLanguages([]);
 			return;
 		}
 
-		// The below function will throw an error if the provided language is
-		// not supported, so we provide fallbacks.
-		// https://github.com/laurent22/joplin/issues/4146
-		const languagesToTry = [
-			v,
-			languageCodeOnly(v),
-		].concat(localesFromLanguageCode(languageCodeOnly(v), this.availableLanguages));
-
-		for (const toTry of languagesToTry) {
-			try {
-				this.session().setSpellCheckerLanguages([toTry]);
-				logger.info(`Set effective language from "${v}" to "${toTry}"`);
-				return;
-			} catch (error) {
-				logger.warn(`Failed to set language to "${toTry}". Will try the next one in this list: ${JSON.stringify(languagesToTry)}`);
-				logger.warn('Error was:', error);
-			}
-		}
-
-		logger.error(`Could not set language to: ${v}`);
+		this.session().setSpellCheckerLanguages(v);
+		logger.info(`Set effective languages to "${v}"`);
 	}
 
 	public get language(): string {
 		const languages = this.session().getSpellCheckerLanguages();
 		return languages.length ? languages[0] : '';
 	}
 
-	public addWordToSpellCheckerDictionary(_language: string, word: string) {
+	public addWordToSpellCheckerDictionary(word: string, _language?: string) {
```

Comment:
Please revert the arguments to the original order. And none should be optional.

Reply:
The same problem as with `setLanguage` function. I changed it, so the `SpellCheckerServiceDriverBase` and `SpellCheckerServiceDriverNative` would have the same arguments, or it causes linting errors. Do you really want this change? `addWordToSpellCheckerDictionary` is not used in any other parts of the code base, so making argument optional, should not raise any issues.

Reply:
I think the reason for avoiding the change is to prevent subtle regressions (both arguments are `string`s and so the compiler won't warn if they're in the wrong order). This could be problematic if another PR is trying to use `addWordToSpellCheckerDictionary` (which might be unlikely).

One option here would be to wrap one of the arguments in an interface (e.g. `interface SpellCheckerLanguage { language: string; }`, defined elsewhere). I don't like this solution, however...

Another option could be to make `_language?` a string array, instead of a `string`.

Reply:
> I think the reason for avoiding the change is to prevent subtle regressions (both arguments are `string`s and so the compiler won't warn if they're in the wrong order). This could be problematic if another PR is trying to use `addWordToSpellCheckerDictionary`.

They was both strings even before I made the change.

I think current implementation is fine. Pass a word to add to dictionary, if other spell checker need to associate word with language, there is option for that.

Reply:
> The same problem as with setLanguage function. I changed it, so the SpellCheckerServiceDriverBase and SpellCheckerServiceDriverNative would have the same arguments, or it causes linting errors.

I just had a look at the code again but I don't understand what you mean. addWordToSpellCheckerDictionary always had (_language, _word) as arguments. If one of them is not needed in a particular context, just ignore it inside the function. Please keep both arguments required.